### PR TITLE
Devops-06 Investigating Prisma 

### PR DESCRIPTION
**Pull request by JonathanBanerjee**
@lgiordani 

Hi Leo, 

In this PR I have been looking into the Prisma schema. I have a better understanding on how it works now. I have been able to generate the prisma client in the `dockerfile` in the `server` however when I run the containers and view the console logs, it looks for `localhost:3002/solutions ` I searched to see if this was mentioned anywhere in the code and I could not locate any mentions. The closest mention was in the `server.js` file where it has `app.get('/solutions', async (_req, res) => {` in `line 29`  However, when I commented this out, it was still searching for `localhost:3002/solutions `  so I am a bit lost as to what may be causing this.

### CODE REVIEW

**Comment by lgiordani** [comment]
@JonathanBanerjee apologies for the long hiatus. Let's start again, as I can't really remember what is going on here. Let's say I have the code of this branch. What is it that you are trying to achieve and what is it that doesn't work?
Thanks

PS: the README file is badly formatted, I recommend you fix that as it's the first thing your user will see.

**Comment by lgiordani** [comment]
Update: I can actually run `db` and `server` and access `localhost:3002/letters` and `localhost:3002/solutions`. All is good there, I think

**Comment by lgiordani** [comment]
OK, I had an in-depth look. My impression is that you are trying to see if you can run something 1) coded in a language you don't really know (Typescript) 2) using a framework (Prisma) you don't know 3) interacting with a technology you don't know (Postgres).
Let me rephrase the question here: what is that you are trying to learn from this exercise?

I ask this because I worked on it 30 minutes, and I was swamped in all possible issues with Prisma (which I don't really know, just like you), version mismatches between client and server, and overall trying to hammer something until it works. What for? Why don't you start with something you coded, using technologies that you understand, and that you can describe?

Just to clarify, my intention is not to shatter your expectations, but I'm concerned that, even if we make it work (and yes, with a lot of hammering we can make it work) you won't really get anything out of it other than a lot of frustration. :scream_cat: Nothing good comes from frustration! :sob: 

You know JS now, right? Why don't you start from scratch with a React application that uses a DB? That way you have code that you wrote and that you know.

Anyway, if you want my assessment on what is going on here, this is it: Prisma is an ORM, so it manages the client, the queries, and the schema of the DB. The command `prisma generate` that you added actually creates JS code that interacts with the DB. However, this poses an issue, as we want to populate the DB from outside, and so we need to match the Prisma schema. Moreover, Prisma has an issue with the prefixes. Apparently it looks for `public.Letters` instead of `Letters`, and a quick search showed me that there might be issues with the versions (https://github.com/prisma/prisma/issues/10587) . However, I don't really know the ORM itself, so I'm assuming a lot of things here.

**Comment by JonathanBanerjee** [comment]
Hey Leo, 

Thank you so much for this! Yes I agree it is quite a steep challenge to tackle right now. Especially as technologies will have updated in the past year. I am in two camps as I like the challenge and being able to overcome it and I feel it would help me gain a better understanding. For example, I'm sure DevOps engineers would have to learn technologies they are not familiar with fairly frequently. However, I am thinking that in terms of a personal project it may be better to start with something a little more simple as you say. 

I'm still working on a few CSS and JavaScript projects at the moment but I hope to work on a few DevOps  projects in the future.
